packages/core/src/chain.ts

```diff
@@ -8,7 +8,7 @@ import { validateSetChainOptions } from './validation'
 import type { WalletState } from './types'
 
 async function setChain(options: {
-  chainId: string | number
+  chainId: string 
```

Comment:
```suggestion
  chainId: string | number
```
So I think that we still want to accept a `number` type here for the `chainId` param.

Then under this line: `const { chainId, chainNamespace = 'evm', wallet: walletToSet } = options`
we want to normalize it as a hex string: 
```javascript
const chainIdHex = typeof chainId === 'number' ? `0x${chainId.toString(16)}` : chainId
```

Then we can refer to `chainIdHex` instead of `chainId` for all comparisons underneath.

You could change the utils functions so that we have a simple function that takes a chainId and ensures it is a hex string and then have another function that takes an array of chains and uses that function to convert all chainId's to hex.